.github/workflows/cmake_build.yml

```diff
@@ -0,0 +1,259 @@
+  standard_tests:
+    name: ${{ matrix.os }} & ${{ matrix.compiler.CC }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-16.04, ubuntu-18.04, 	ubuntu-20.04, macos-10.15]
+        compiler:
+          - { CC: gcc, CXX: g++ }
+          - { CC: clang, CXX: clang++ }
+
+    steps:
+      - name: Get Source
+        uses: actions/checkout@v2
+  
+      - name: "Linux: Run Apt-Get"
+        if: startsWith(matrix.os, 'ubuntu') && matrix.os != 'ubuntu-20.04'
+        run: sudo apt-get install elfutils libzstd1-dev
+  
+      - name: "Linux (Ubuntu 20.04): Run Apt-Get"
+        if: matrix.os == 'ubuntu-20.04'
+        run: sudo apt-get install elfutils libzstd-dev
+  
+      - name: "Setup common ENV"
+        uses: allenevans/set-env@v1.1.0
+        with:
+          CC: ${{ matrix.compiler.CC }}
+          CXX: ${{ matrix.compiler.CXX }}
+          VERBOSE: 1
+          ENABLE_CACHE_CLEANUP_TESTS: true
+          CTEST_OUTPUT_ON_FAILURE: ON
+      
+      - name: Build and Test
+        shell: bash
+        run: ci/build
+  
+      - name: Collect testdirs from failed tests
+        if: failure()
+        run: zip -r testdirs.zip ${{ matrix.config.BUILDDIR }}/testdir.* -x testdir.failed
+
+      - name: Upload testdirs from failed tests
+        if: failure()
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.config.name }} - testdirs.zip
+          path: testdirs.zip
+
+      # Does it make sense to do this on every matrix element?
+      - name: "Build binary and verify it"
+        run: ci/build-and-verify-package
+        # ToDo: upload testdirs in case of errors
+
+      - name: Upload binary
+        uses: actions/upload-artifact@v2
+        with:
+          # Does it make sense to upload all of them?
+          # What is even the use case? People using specific versions without compiling on their own.
+          name: ${{ matrix.os }}-${{ matrix.compiler.CXX }}-ccache.tar.gz
+          path: build_package_dir_test/ccache-binary.tar.gz
+
+      # Does it make sense to do this on every matrix element?
+      - name: "Create source package and verify it"
+        run: ci/build-and-verify-package-source
+        # ToDo: upload testdirs in case of errors
+
+      - name: Upload source zip
+        # Uploading once is enough
+        if: matrix.os == 'ubuntu-20.04' && matrix.CC == 'gcc'
+        uses: actions/upload-artifact@v2
+        with:
+          name: ${{ matrix.os }}-${{ matrix.compiler.CXX }}-ccache-src.tar.gz
+          path: build_package_source_dir_test/ccache-src.tar.gz
+
+  # These mimic the old Travis tests as far as not replaced by standard_tests:
+  specific_tests:
+    name: ${{ matrix.config.name }}
+    runs-on: ${{ matrix.config.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        config:
+          - {
+            # Job 1: Build (in source directory) on Linux with native GCC in release build mode
```

Comment:
I added job numbers in the .travis.yml comments to make it easier to correlate the jobs to the Travis-CI jobs, but since GitHub Actions jobs are unnumbered we should just remove them since they don't serve a purpose.

Reply:
The purpose is rather temporary indeed. I wanted to have them easily comparable to travis.
(Review, keeping in sync until full switch, etc)